### PR TITLE
Parties

### LINKED ISSUE #23: Parties / Gatherings / Multiple Wishlists
In the main description it says: 

> A user can build a wishlist and everybody else can check things off of it without the user being able to see.

Currently the functionality allows a user to see another's wish list and check a box.  The only other action item is the "leave this wishlist."

Wouldn't it be cool to see other user wishlists in the same party/gathering?

Using the `parties` name, a db structure could look like this?

`parties`
`parties_users`
`parties_wishlists`

### PR DESCRIPTION
**Pull request by panreach**
Closes #23 

- [x] basic CRUD for parties
- [x] Fix auto add user to party after login/create-account from share link
- [ ] Add phpunit tests
- [ ] Allow fulfillment viewing from parties page with mass toggle?
- [ ] Add easy way for legacy users to add their non-party wishlist to a party.
- [ ] ...

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -19,6 +20,12 @@
 
 Route::middleware('auth')->prefix('/app')->group(function () {
     Route::get('/', AppController::class)->name('app');
+    
+    Route::get('/parties', [PartyController::class, 'index'])->name('parties.index');
+    Route::get('/parties/create', [PartyController::class, 'create'])->name('parties.create');//->can('create', Party::class); // @TODO can't figure out why unauthorized
```
Comment (panreach) [inline routes/web.php]: @imacrayon This one throws me.  Is my policy wrong or something else?

Comment (panreach) [reply routes/web.php]: @TODO can't figure out why unauthorized

Comment (imacrayon) [reply routes/web.php]: Make sure you have `use App\Models\Party` at the top of the file. It’s fails silently if you don’t have the class imported.

File: database/migrations/2023_11_07_011743_create_parties_table.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        Schema::create('parties', function (Blueprint $table) {
+            $table->id();
+            $table->foreignId('user_id_created_by')->constrained('users');
+            $table->string('name');
+            $table->text('description', 65535)->nullable();
+            $table->string('address', 255)->nullable();
+            $table->float('lat', 10, 6)->nullable();
+            $table->float('lng', 10, 6)->nullable();
+            $table->dateTime('start_datetime')->nullable();
+            $table->dateTime('end_datetime')->nullable();
```
Comment (imacrayon) [inline database/migrations/2023_11_07_011743_create_parties_table.php]: I wonder if we should hold off on the time/location stuff for now. That will get tricky to build and leaving it out will make Parties more general purpose. For example, my family would probably just reuse the same party "group" each year, so it wouldn't need any time/location data. But if someone really wanted to provide information for a one-off party they could include that time/location info in the `description` field.

File: routes/web.php (hunk 1 above)
Comment (panreach) [reply routes/web.php]: Doh... I didn't have `use App\Models\Party;`.

File: database/migrations/2023_11_07_011743_create_parties_table.php (hunk 2 above)
Comment (panreach) [reply database/migrations/2023_11_07_011743_create_parties_table.php]: That's a very good point.  I've added `use_address_bool/use_dates_bool` fields and have defaulted them to false.

**Comment by imacrayon** [comment]
Hey @panreach I realized we pushed quite a few changes lately that conflict with some of the stuff you're working on in this PR. It looks like the on-boarding process for parties might require some reworking to fit with some of new layout and guest user features specifically. I think parties would still be a nice feature to support, so if you don't mind, I'll work on resolving the conflicts in this PR and try to sort out at least a basic version we can ship soon. It feels like this is 80% of the way there, thanks for all your work on it so far!

**Comment by panreach** [comment]
> I think parties would still be a nice feature to support, so if you don't mind, I'll work on resolving the conflicts in this PR and try to sort out at least a basic version we can ship soon.

Go for it @imacrayon

**Pull request merged**